### DISCUSSION TITLE
What is the library/upload folder used for?

### DISCUSSION THREAD

**Discussion by Tiypoh** [immich-app/immich#12591] [Q&A]
I was going though my server trying to clear up space and found that out of the 2.1 TB on my server, 446 GB was in the /library/upload/ folder, I was wondering what its used for and if theres any way to clear up that space since as of right now Ive only got 1 TB of images uploaded, but twice the space is being taken up in the /immich-app/ folder.

**Comment by bo0tzz** [comment]
When files are uploaded they go into that folder. If storage template is enabled then after the processing they get moved into /library/library

**Reply by Tiypoh** [reply]
Thats what I figured, but then why are they still there? I havent uploaded anything in a week or so and theyve all finished uploading of course. And if its just an error, can I delete the files in there? Futhermore its full of folders with 2 digit hexadecimal values, what are those for, and do I just have to sort through each one and delete the files in each?

**Reply by smailpouri** [reply]
i'm having the same issue. I have roughly the same amount of data as on my library folder, I run the `Storage Migration` process but that does not change anything. 

Is it safe to delete the upload folder?

<img width="235" alt="Screenshot 2024-09-26 at 9 56 29 PM" src="https://github.com/user-attachments/assets/70c4cf57-f26b-44bb-a860-e8c4fe3ae443">

_
It also seems slightly off from what the web interface states:
<img width="641" alt="Screenshot 2024-09-26 at 9 59 03 PM" src="https://github.com/user-attachments/assets/487f9ddf-8d9c-47ed-badd-35b58a5b1636">

**Comment by michalvankodev** [comment]
I have a similar issue. Most of the disk space is taken up by `/upload` folder.

![image](https://github.com/user-attachments/assets/9c6c9420-43be-464b-bf1f-48a535d0656b)

When I take a look at the storage it reports only 110 GB 
![image](https://github.com/user-attachments/assets/98e554c1-ea84-40c2-8d85-36ace2d42cf8) which I would also consider wrong as the encoded-video and thumbs should be also included.

**Comment by Tiypoh** [comment]
@bo0tzz Can you please give some insight into all of this, my Immich install is currently taking up 2.7TB after only uploading probably 1.5TB at most, with most of that in the upload folder. Can you let us know if we can clean this up please.

**Reply by bo0tzz** [reply]
If you've run the storage template job without any errors then those files in upload/ should be ok to delete.

**Reply by Tiypoh** [reply]
Okay thank you very much!

**Reply by Tiypoh** [reply]
For anyone else doing this: I tested it, dont delete the whole upload folder, just the userid folders within it, otherwise immich wont start
